spring-integration-core/src/main/java/org/springframework/integration/dispatcher/BroadcastingDispatcher.java

```diff
@@ -158,12 +159,16 @@ public boolean dispatch(Message<?> message) {
 			throw new MessageDispatchingException(message, "Dispatcher has no subscribers");
 		}
 		int sequenceSize = handlers.size();
+		Message<?> messageToSend = message;
+		UUID sequenceId = null;
+		if (this.applySequence) {
```

Comment:
Otherwise we call `message.getHeaders().getId()` for each `MessageHandler` in the loop.